src/storage/src/table/cell_based_table.rs

```diff
@@ -417,113 +486,78 @@ impl<S: StateStore> CellBasedTable<S> {
             (key_range.start_bound(), key_range.end_bound())
         );
 
-        CellBasedTableRowIter::new_with_bounds(
-            &self.keyspace,
-            self.mapping.clone(),
-            key_range,
-            epoch,
-        )
-        .await
-    }
-
-    pub fn schema(&self) -> &Schema {
-        &self.schema
+        self.batch_iter_with_encoded_key_range(key_range, epoch)
+            .await
     }
 }
 
-fn generate_column_id(column_descs: &[ColumnDesc]) -> Vec<ColumnId> {
-    column_descs.iter().map(|d| d.column_id).collect()
-}
+const STREAMING_ITER_TYPE: bool = true;
+const BATCH_ITER_TYPE: bool = false;
 
-// (st1page): Maybe we will have a "ChunkIter" trait which returns a chunk each time, so the name
-// "RowTableIter" is reserved now
-pub struct CellBasedTableRowIter<S: StateStore> {
+/// [`CellBasedIter`] iterates on the cell-based table.
```

Comment:
No need to link to itself? 🤪

Reply:
I'm afraid the clippy will complain about this. 😁